### PR TITLE
Created a base for the ui using the command pattern

### PR DESCRIPTION
**Pull request by SzymonSztyrmer**
PR has base ui implementation with no functionality.
Set up using the command pattern.
More functionality will be added soon

### CODE REVIEW

**Review by Bmalone100 — APPROVED** [review]
Tactful validation as always XD

**Review by SzymonSztyrmer — COMMENTED** [review]
This update moves the cli into its own package along with the command
Sets up a ui for the character creation(name, race and class)
Things on this branch are dependent on chief's stuff to work
Weapon and armour is yet to be added.
Will like refactor the validation to shorten the code base.

File: src/main/java/com/seventhtill/ui/Cli.java [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+        Race characterRace = null;
+        AbstractFactory raceFactory = FactoryProducer.getFactory(
+                "human");
+        System.out.println("Do you want to be a standard human?:\n" +
+                "1) Standard Human.\n" +
+                "2) Cancel\n");
+        String userInput = this.scanner.nextLine();
+        switch (userInput) {
+            case "1":
+                characterRace = raceFactory.getRace("highelf");
```
Comment (LouiseMadden) [inline src/main/java/com/seventhtill/ui/Cli.java:283]: Shouldn't this be standard human?

**Review by LouiseMadden — CHANGES_REQUESTED** [review]
Just one little thing but otherwise looks good to me, need to fix the standard human in Cli.java

File: src/main/java/com/seventhtill/ui/Cli.java (hunk 1 above)
Comment (SzymonSztyrmer) [reply src/main/java/com/seventhtill/ui/Cli.java:283]: oh ya
that's a copy/paste blunder will fix

**Review by SzymonSztyrmer — COMMENTED** [review]
Big boi refactor to throw out duplicated code

**Review by DylanPKing — APPROVED** [review]

**Review by LouiseMadden — APPROVED** [review]
LGTM

**Pull request merged**